src/server/auth/auth.service.ts

```diff
@@ -0,0 +1,17 @@
+import { UserService } from './../user/user.service';
+import { Injectable } from "@nestjs/common";
+import { StringLiteralType } from 'typescript';
+
+@Injectable() 
+export class AuthService{
+    constructor(private userService: UserService) {}
+
+    async validateUser(username : string, password: string): Promise<any> {
+        const user = await this.userService.findUser(username)
+
+        if(user && user.password === password ) {
```

Comment:
we store password hashes in our database, so we can't simply check if they're equal